File: app/src/main/java/com/junlong0716/basemvp/BasePresenter.java
package com.junlong0716.basemvp;

import android.os.Bundle;

import javax.inject.Inject;

public abstract class BasePresenter<A extends BaseActivity, M extends BaseModel> {
    protected A mActivity;
    @Inject
    protected M mModel;

    public void setActivity(A activity) {
        this.mActivity = activity;
    }

    @Inject
    public void setPresenter() {
        this.mModel.setPresenter(this);
    }

    public void onCreate(Bundle savedInstanceState) {

    }
}

File: app/src/main/java/com/junlong0716/basemvp/BasePresenter.kt
package com.junlong0716.basemvp
import android.content.Context

class BasePresenter<M : BaseModel, V : BaseView> : IPresenter {
    private var mModel: M? = null

    private var mView: V? = null

    val viewContext: Context?
        get() = null

    constructor(model: M, view: V) {
        mModel = model
        mView = view
    }

    constructor(view: V) {
        mView = view
    }

    override fun onDestroy() {
        if (mModel != null) {
            mModel!!.onDestroy()
        }
        this.mModel = null
        this.mView = null
    }
}
